impeller/renderer/backend/vulkan/binding_helpers.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <vector>
+
+#include "impeller/renderer/backend/vulkan/context_vk.h"
+#include "impeller/renderer/command.h"
+#include "impeller/renderer/compute_command.h"
+
+namespace impeller {
+
+std::vector<vk::DescriptorSet> AllocateAndBindDescriptorSets(
+    const ContextVK& context,
+    const std::shared_ptr<CommandEncoderVK>& encoder,
+    const std::vector<Command>& commands);
+
+std::vector<vk::DescriptorSet> AllocateAndBindDescriptorSets(
```

Comment:
Ooh, the overload is unfortunate. Trying to figure out how we can dry up this code. But perhaps I am overthinking it.

Reply:
I could make this a template monstrosity but I think lets take the win for now from sharing BindTexture/BindBuffer. Long term, I think we fix this by breaking apart Command/ComputeCommand and storing the encoding data more generically

Reply:
Yeah, I was going to suggest reworking it but realized thats going to be non-trivial. Later, if at all.